controllers/discovery/initDiscoveryTopicData.js

```diff
@@ -12,21 +13,22 @@ const InitDiscoveryTopicData = async (req, res) => {
   page = parseInt(page);
 
   const userId = req.userId;
+  const anonymousUser = await UsersFunction.findAnonymousUserId(User, userId, {raw: true});
 
   try {
     let totalDataQuery = `SELECT 
                               count(A.topic_id) as total_data
                           FROM topics A`;
 
     let suggestedTopicsQuery = `SELECT 
-                C.*, 
+                C.*,
                 A.topic_id, 
                 COUNT(*) as common,
                 A.user_id as user_id_follower
             FROM user_topics A 
             INNER JOIN user_topics B 
                 ON A.topic_id = B.topic_id 
-                AND A.user_id = :userId
+                AND (A.user_id = :userId OR A.user_id = :anonymousUserId)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The SQL query has been modified to include conditions for anonymous users. Ensure that the `anonymousUserId` is properly sanitized to prevent SQL injection attacks. Also, make sure that the logic of the query still holds with the addition of the new condition.
```diff
- AND A.user_id = :userId
+ AND (A.user_id = :userId OR A.user_id = :anonymousUserId)
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->